ocaml/xapi/repository_helpers.ml

```diff
@@ -684,27 +698,56 @@ let get_cachedir repo_name =
         let msg = Printf.sprintf "Not found cachedir for repository %s" repo_name in
         raise Api_errors.(Server_error (internal_error, [msg])))
 
-let with_local_repository ~__context f =
+let get_enabled_repositories ~__context =
+  let pool = Helpers.get_pool ~__context in
+  Db.Pool.get_repositories ~__context ~self:pool
+
+let get_remote_repository_name ~__context ~self =
+    !Xapi_globs.remote_repository_prefix ^ "-" ^ (Db.Repository.get_name_label ~__context ~self)
```

Comment:
The `name_label` field is now used as an important key that is used internally in names and URLs, where previously it wasn't. The datamodel defines it as a RW field along with `name_description`, and can be used by a client to put in strings without validation and at any time. If we are going to use the `name_label` as part of paths in the file system or URLs, then we need to carefully constrain them (e.g. check for bad characters) and encode where needed. We'd also need to handle changes to the field.

An alternative could be to use the UUID instead, which is fixed and has a known format.

Reply:
It's good idea to use UUID. I just appended a new commit for that.